src/textarea/textarea.tsx

```diff
@@ -9,25 +9,127 @@ interface Props {
+                        <circle
+                            id="colored"
+                            cx="50%"
+                            cy="50%"
+                            r={this.state.radius}
+                            className={this.state.stroke}
+                            style={{ strokeDasharray: this.state.strokeDasharray }}
+                        />
+                        <text
+                            className={this.state.stroke}
+                            x="50%"
+                            y="50%"
+                            text-anchor="middle"
+                            stroke-width="1px"
+                            dy=".3em"
+                        >
+                            {this.state.limit}
+                        </text>
+                    </svg>
+                )}
+            </div>
+        );
+    }
+    private onChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        this.props.onChange(event.target.value);
+        const { minRows, maxRows } = this.state;
+
+        const currentRows = this.getCurrentRows(event, minRows, maxRows);
+        this.setState({
+            value: event.target.value,
+            rows: currentRows < maxRows ? currentRows : maxRows,
+            limit: this.state.characterLimit - event.target.value.length,
+            stroke: this.state.characterLimit - event.target.value.length <= 0 ? 'error-text' : 'normal-text',
+            strokeDasharray: `${this.getPercentage(event)} 999`,
+        });
+    }
+
+    private getCurrentRows(event: React.ChangeEvent<HTMLTextAreaElement>, minRows: number, maxRows: number) {
+        if (!this.props.isCharacterLimitEnabled) {
+            return this.state.rows;
+        }
+        const textareaLineHeight = 14;
+
+        const previousRows = event.target.rows;
+        event.target.rows = minRows;
 
-const Textarea: React.FunctionComponent<Props> = (props) => {
-    const onChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => props.onChange(event.target.value);
-    return (
-        <div className="textarea-component">
-            <textarea
-                id={props.id}
-                className={`form-control ${props.className || ''}`}
-                rows={props.rows || 3}
-                onBlur={props.onBlur}
-                onChange={onChange}
-                placeholder={props.placeholder}
-                data-qa={`textarea--element-${props.id}`}
-                value={props.value}
-                disabled={props.disabled}
-            />
-        </div>
-    );
-};
+        const currentRows = Math.floor(event.currentTarget.scrollHeight / textareaLineHeight);
+
+        if (currentRows === previousRows) {
+            event.target.rows = currentRows;
+        }
+
+        if (currentRows >= maxRows) {
+            event.currentTarget.rows = maxRows;
+            event.currentTarget.scrollTop = event.currentTarget.scrollHeight;
```

Comment:
What is the purpose here?

Reply:
this is to make the Textarea component flexible by growing it according to the size

Reply:
No this actually scrolls on the given element: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop
Doesn't that happen automatically?

Reply:
yes when we hit 10 rows add a scroll after that